pygmt/base_plotting.py

```diff
@@ -55,1591 +40,20 @@ def _preprocess(self, **kwargs):  # pylint: disable=no-self-use
-                file_context = lib.virtualfile_from_matrix(data)
-            elif kind == "vectors":
-                file_context = lib.virtualfile_from_vectors(x, y, z)
-
-            with file_context as fname:
-                arg_str = " ".join([fname, build_arg_string(kwargs)])
-                lib.call_module("contour", arg_str)
-
-    @fmt_docstring
-    @use_alias(
-        R="region",
-        J="projection",
-        Jz="zscale",
-        JZ="zsize",
-        B="frame",
-        L="map_scale",
-        Td="rose",
-        Tm="compass",
-        U="timestamp",
-        V="verbose",
-        X="xshift",
-        Y="yshift",
-        p="perspective",
-        t="transparency",
-    )
-    @kwargs_to_strings(R="sequence", p="sequence")
-    def basemap(self, **kwargs):
-        """
-        Plot base maps and frames for the figure.
-
-        Creates a basic or fancy basemap with axes, fill, and titles. Several
-        map projections are available, and the user may specify separate
-        tick-mark intervals for boundary annotation, ticking, and [optionally]
-        gridlines. A simple map scale or directional rose may also be plotted.
-
-        At least one of the options *frame*, *map_scale*, *rose* or *compass*
-        must be specified.
-
-        Full option list at :gmt-docs:`basemap.html`
-
-        {aliases}
-
-        Parameters
-        ----------
-        {J}
-        zscale/zsize : float or str
-            Set z-axis scaling or z-axis size.
-        {R}
-        {B}
-        map_scale : str
-            ``'[g|j|J|n|x]refpoint'``
-            Draws a simple map scale centered on the reference point specified.
-        rose : str
-            Draws a map directional rose on the map at the location defined by
-            the reference and anchor points.
-        compass : str
-            Draws a map magnetic rose on the map at the location defined by the
-            reference and anchor points
-        {U}
-        {V}
-        {XY}
-        {p}
-        {t}
-        """
-        kwargs = self._preprocess(**kwargs)
-        if not args_in_kwargs(args=["B", "L", "Td", "Tm"], kwargs=kwargs):
-            raise GMTInvalidInput(
-                "At least one of frame, map_scale, compass, or rose must be specified."
-            )
-        with Session() as lib:
-            lib.call_module("basemap", build_arg_string(kwargs))
-
-    @fmt_docstring
-    @use_alias(
-        R="region",
-        J="projection",
-        D="position",
-        F="box",
-        S="style",
-        U="timestamp",
-        V="verbose",
-        X="xshift",
-        Y="yshift",
-        t="transparency",
+    from pygmt.src import (  # pylint: disable=import-outside-toplevel
+        basemap,
+        coast,
+        colorbar,
+        contour,
+        grdcontour,
+        grdimage,
+        grdview,
+        image,
+        inset,
+        legend,
+        logo,
+        meca,
+        plot,
+        plot3d,
+        text,
     )
```

Comment:
As mentioned in https://github.com/GenericMappingTools/pygmt/pull/808#discussion_r570575569, could you delete `base_plotting.py` (can't believe I'm saying this!) and move this import chunk to `figure.py`? This will fix the pylint `R0903: Too few public methods (0/2) (too-few-public-methods)` recommendation. Feel free to copy any of the docstrings if needed.

Reply:
@weiji14 I just moved this import string to `figure.py` and removed imports for `base_plotting.py`, but when I ran `make test` I had 196 test failures. I'm not sure what the need to `_preprocess` is, but it seems to play some sort of role.

Reply:
Just double checking that you placed the imports under `class Figure():`? I.e. indented 4 spaces? The `_preprocess` class function activates the correct figure to plot in.

Reply:
...nope 🤦‍♂️

Removed `base_plotting.py` in 5ac2e07 and actually put the imports in `Figure()` and it worked (or at least passed all of the tests)!

Reply:
> As mentioned in [#808 (comment)](https://github.com/GenericMappingTools/pygmt/pull/808#discussion_r570575569), could you delete `base_plotting.py` (can't believe I'm saying this!) and move this import chunk to `figure.py`? This will fix the pylint `R0903: Too few public methods (0/2) (too-few-public-methods)` recommendation. Feel free to copy any of the docstrings if needed.

We should be careful with the changes. There are two different `_preprocess()` functions in `Figure()` and `BasePlotting`. If I understand it correctly, the `_preprocess` function if `BasePlotting` override the one in `Figure()`.

Reply:
@seisman The tests are passing; is there an issue with deleting base_plotting.py?

Reply:
> We should be careful with the changes. There are two different `_preprocess()` functions in `Figure()` and `BasePlotting`. If I understand it correctly, the `_preprocess` function if `BasePlotting` override the one in `Figure()`.

I feel like the `_preprocess` in base_plotting.py was just a dummy placeholder, and the one in figure.py is the actual one used. So probably the other way around? Might need to read up on subclassing, I'm not terribly good at object orientated conventions.

Reply:
I thought `BasePlotting()` is a subclass of `Figure()`, now it makes sense to me.